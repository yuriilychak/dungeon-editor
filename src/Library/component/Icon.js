import React from "react";
import { makeStyles } from "@material-ui/styles";

const useStyles = makeStyles({
    root: {
        width: 16,
        height: 16,
        paddingRight: "0 !important",
        fill: "#ffffff",
        userDrag: "none",
        userSelect: "none"
    }
});

const Icon = ({name}) => (
    <img src={`${process.env.PUBLIC_URL}/icon/${name}.svg`} className={useStyles().root} alt={"empty Icon"} draggable={false}/>
);

export default Icon;
