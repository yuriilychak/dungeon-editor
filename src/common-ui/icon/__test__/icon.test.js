import React from "react";
import { createMount } from "@material-ui/core/test-utils";
import Icon from "../icon";

describe("Icon test",()=> {

    it('default snapshot', () => {
        const wrapper = createMount()(
            <Icon name="test" />
        );
        expect(wrapper.html()).toMatchSnapshot();
    });
});
