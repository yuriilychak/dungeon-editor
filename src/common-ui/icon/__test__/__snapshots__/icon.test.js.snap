// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon test default snapshot 1`] = `"<img src=\\"undefined/icon/test.svg\\" class=\\"icon-root\\" alt=\\"empty Icon\\" draggable=\\"false\\">"`;

exports[`Icon test with icon default snapshot 1`] = `"<img src=\\"undefined/icon/test.svg\\" class=\\"icon-root\\" alt=\\"empty Icon\\" draggable=\\"false\\">"`;
